### PR TITLE
integrate gocyclo

### PR DESCRIPTION
**Pull request by erikh**
errcheck was removed because it was just too much atm. I will address this later.

Anything `> 15` on gocyclo was resolved. PTAL.

### CODE REVIEW

**Comment by erikh** [comment]
@dseevr @yuva29 @vvb btw, please review this carefully, a lot of refactors were done.

**Comment by dseevr** [comment]
LGTM

**Comment by erikh** [comment]
PTAL @yuva29 @vvb

File: apiserver/daemon.go [hunk 1]
```diff
@@ -676,76 +652,118 @@ func (d *DaemonConfig) handleRemove(w http.ResponseWriter, r *http.Request) {
...
 		}
 
 		if !exists {
-			etcdRemove()
+			d.removeVolume(req, vc)
 			return errors.NotExists
 		}
 
-		return complete()
+		return d.completeRemove(req, vc)
```
Comment (vvb) [inline apiserver/daemon.go]: @erikh may be we can rewrite this as -
```
if err == errors.NoActionTaken || exists {
	return d.completeRemove(req, vc)
}
  		  
d.removeVolume(req, vc)
return errors.NotExists
```

**Comment by vvb** [comment]
LGTM

**Pull request merged**